### DISCUSSION TITLE
`noConsole` vs `noConsoleLog`?

### DISCUSSION THREAD

**Discussion by drwpow** [biomejs/biome#2577] [Ideas]
Biome has a `noConsoleLog` rule, as opposed to ESLint’s `no-console` rule. The former _only_ warns/errs on `console.log()` literally. But the `no-console` rule warns/errs on `console.error`, `console.warn`, and `console.debug` as well:

**Biome** `noConsoleLog`

![CleanShot 2024-04-23 at 16 10 15@2x](https://github.com/biomejs/biome/assets/1369770/282e2ad1-9cef-4942-974a-2e86dc2629b5)

**ESLint** `no-console`

![CleanShot 2024-04-23 at 16 18 31@2x](https://github.com/biomejs/biome/assets/1369770/60fcb73d-ca6d-439c-a530-1887bd6527f4)

My question is: **should Biome get additional lint rules like `noConsoleError` to handle the other cases? Or does Biome just need a `noConsole` rule to warn/err on all of them collectively?** 

Or is the single `noConsoleLog` rule intentional, going off the assumption that it’s mostly `console.log` that’s used by accident and the other ones are intentional, so there’s no need to make rules for them?

**Comment by ematipico** [comment]
@anonrig what's your take here? I would love to have your thoughts here.

**Comment by Conaclos** [comment]
Usually developers use `console.log` to debug their code.
Other methods such as `console.info`, `console.warn` and `console.error` are used to log information, warnings, and errors.

I implemented [noConsoleLog](https://biomejs.dev/linter/rules/no-console-log) to warn against debugging code.
Therefore, the rule doesn't warn about other uses of console.

However, some users/contributors, such as @anonrig, want to warn against other uses of `console`.
I am not sure of the reasoning behind this.
@anonrig has implemented [noConsole] (https://biomejs.dev/linter/rules/no-console).
The plan is to add an option to whitelist some uses.
We could deprecate `noConsoleLog` once `noConsole` is stabilized.

**Reply by Conaclos** [reply]
@drwpow what behavior do you want?

**Reply by drwpow** [reply]
`noConsole` would work for me! That would probably be simplest as it maintains parity with `no-console`.

Just speaking personally, I have 2 main usecases where I enforce this rule:

1. I’m writing browser code, and it’s just bad form to clutter up the user’s console with non-essential information (i.e. devs should have an explanation why logging to the browser console is necessary)
2. I’m writing a Node.js CLI utility, and I need to be conscious of when/where I’m logging to the console because I may need to preserve `stdout` as the output of a function (and not just diagnostics).

In both scenarios, I want to catch `log`/`debug`/`warn`/`error` indiscriminately, which `noConsole` would do nicely for. Though I’m sure there are some usecases where someone may be concerned with individual logging methods, I personally don’t need it.

**Comment by Netail** [comment]
Seems to be implemented: https://biomejs.dev/linter/rules/no-console/ :)
